### DISCUSSION TITLE
About populateState method on component ListModel models

### DISCUSSION THREAD

**Discussion by joomdonation** [joomla/joomla-cms#33765] [General]
Does anyone knows why we have the code to populate model states like in this block of code https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_modules/src/Model/ModulesModel.php#L92-L96 ?

- We do not have input variable such as `filter_search`, `filter_position`.... to get data to set value for model states (note that we are using search tool, so the values for these input are stored in `filter`, not separate inputs like that.
- These model states, if set, will be over-write by **populateState** method from parent class immediately https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/MVC/Model/ListModel.php#L500-L510

So I don't see any reasons the code is needed inside our model classes. Or I'm wrong?

**Comment by Fedik** [comment]
looks like it some legacy, maybe Joomla 2, look in to `ListModel::getUserStateFromRequest()`:

https://github.com/joomla/joomla-cms/blob/eddd451cebfba57cd7884008080838ba9e4119b4/libraries/src/MVC/Model/ListModel.php#L687-L699

**Reply by joomdonation** [reply]
I remember code like that is needed in Joomla 3 because we still had to support Hathor template (which is not using search tools). I thought that we do not these lines of code in Joomla 4, but there is still some part in our codebase need these lines of code to initialize model states, for example https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_cpanel/src/Controller/DisplayController.php#L81-L82

Of course, we have a different ways to initialize model states. But I'm unsure if the project wants this kind of clean up at this state :)

**Reply by richard67** [reply]
Possibly that stuff comes up again by accident with an upmerge after a clean up.
